examples/custom-xviz-layers/src/app.js

```diff
@@ -0,0 +1,172 @@
+
+// __IS_STREAMING__ and __IS_LIVE__ are defined in webpack.config.js
+const exampleLog = require(__IS_STREAMING__
+  ? './log-from-stream'
+  : __IS_LIVE__
+    ? './log-from-live'
+    : './log-from-file').default;
+
+class Example extends PureComponent {
+  state = {
+    log: exampleLog,
+    settings: {
+      viewMode: 'PERSPECTIVE',
+      showTooltip: false
+    }
+  };
+
+  componentDidMount() {
+    this.state.log.on('error', console.error).connect();
+  }
+
+  _onSettingsChange = changedSettings => {
+    this.setState({
+      settings: {...this.state.settings, ...changedSettings}
+    });
+  };
+
+  render() {
+    const {log, settings} = this.state;
+
+    return (
+      <div id="container">
+        <div id="control-panel">
+          <XVIZPanel log={log} name="Metrics" />
+          <hr />
+          <XVIZPanel log={log} name="Camera" />
+          <hr />
+          <Form
+            data={APP_SETTINGS}
+            values={this.state.settings}
+            onChange={this._onSettingsChange}
+          />
+          <StreamSettingsPanel log={log} />
+        </div>
+        <div id="log-panel">
+          <div id="map-view">
+            <LogViewer
+              log={log}
+              mapboxApiAccessToken={MAPBOX_TOKEN}
+              mapStyle={MAP_STYLE}
+              car={CAR}
+              xvizStyles={XVIZ_STYLE}
+              showTooltip={settings.showTooltip}
+              viewMode={VIEW_MODE[settings.viewMode]}
+              customXVIZLayers={[
+                {
+                  // {string} id - passed into layer
+                  id: 'dual',
+                  //
+                  extendPrimitive: 'path',
+                  layerClass: LaneLayer,
+                  /* 
+                  * Determines whether the layer handler should be used for a specific stream
+                  * @param {string} streamName
+                  * @param {Object} streamMetadata
+                  */
+
+                  streamMatch: (_, streamMetadata) => {
+                    //
+                    return streamMetadata.primitive_type === 'polyline';
+                  },
+                  /* 
+                  * Defines props to spread into `this.getSublayerProps` in the XVIZLayer
+                  * @param {Object} xvizLayerProps - all props passed into XVIZLayer
+                  * @param {Object} primitiveLayerProps - all layer-type props that are used for the primitive layer
+                  * @param {Object} state - the XVIZLayer state
+                  */
+                  getSubProps: ({xvizLayerProps, primitiveLayerProps, state}) => {
+                    return {
+                      id: 'path-dual',
+                      getWidth: [0.1, 0.1, 0.1],
+                      getPath: f => f.vertices,
+                      getDashArray: () => [1, 2],
+                      getDashArray2: () => [2, 4],
+                      getColor2: state.layerProps.getColor,
```

Comment:
The point here is the ability to use the normal style handling correct.

Reply:
what do you mean exactly?